### PR TITLE
Fix output file functionality in text format selection

### LINKED ISSUE #20: Fix `pr --output ./output.txt` Functionality
#### Description
The `pr --output ./output.txt` command is currently not working as expected. Previously, the tool allowed users to specify output formats like JSON and XML, but this functionality has been lost for the default format This issue aims to restore and fix the output functionality to support both file output and format selection.

#### Problem
- The `--output` flag does not currently write to the specified file (`./output.txt`) when the format is text (the default one).
- The ability to select output formats (e.g., JSON, XML) is missing.

#### Expected Behavior
- Users should be able to specify an output file using `--output ./output.txt`.
- Users should be able to select an output format (e.g., JSON, XML) using a `--format` flag.
- If no format is specified, the default format should be used.

#### TODO:
1. **Fix Output File Functionality**  
   - [ ] Ensure `--output ./output.txt` writes results to the specified file.  

2. **Testing**  
   - [ ] Add tests for `--output` functionality:  
     - Verify file creation and content.  
     - Test with different file paths and formats.  
   - [ ] Add tests for `--format` functionality:  
     - Verify JSON and XML output.  
     - Test edge cases (e.g., invalid formats).  
   - [ ] Ensure backward compatibility (default behavior remains unchanged).  

---

### Example Usage

```bash
# Write output to a file in default format
pr --output ./output.txt
```

### PR DESCRIPTION
**Pull request by zyadtaha**
Fix #20 
- Ensured `--output ./output.txt` writes results to the specified file.
- Added tests for output in text and JSON formats, and also for invalid formats.

### CODE REVIEW

**Comment by Ahmedhossamdev** [comment]
Hey @zyadtaha  

Can you please resolve the conflicts?

**Comment by zyadtaha** [comment]
Done. @Ahmedhossamdev

**Comment by Ahmedhossamdev** [comment]
Great work @zyadtaha  🔥🔥

File: pkg/printer/printer_test.go [hunk 1]
```diff
@@ -235,6 +235,59 @@ func TestPrintProjectStructure(t *testing.T) {
...
+			t.Fatalf("Failed to read output file: %v", err)
+		}
+
+		if len(fileContent) == 0 {
+			t.Errorf("Output file is empty")
+		}
+	})
+
+	// Test JSON output file
+	t.Run("JSONOutputFile", func(t *testing.T) {
```
Comment (Ahmedhossamdev) [inline pkg/printer/printer_test.go:260]: In the feature we can remove the fmt.printF
